src/std/sugar.ss

```diff
@@ -300,3 +301,49 @@
 
   ((_ (v . more) (acc ...) . previous) ; previous is not set after <> was replaced
    (~chain-aux-diamond more (acc ... v) . previous)))
+
+;; is converts a given value into a predicate testing for the presence of the
+;; given value. Optionally a transforming procedure can prefix the value, which
+;; can in this case also be a procedure. This allows to 'get' a value out of a
+;; compound data structure before comparison (first map, then test).
+;; For numbers, char and string specialized procedures are used automatically
+;; if passed to the macro as value and not as variable. Alternatively, the
+;; test: keyword can be used to supply a test, the default is equal?.
+;;
+;; Example:
+;;  (find (is cdr 5) '((a . 2) (b . 5) (c . 6)))
+;; => (b . 5)
+;;
+;;  (filter (is file-type 'regular) (directory-files))
+;; => ("Documents" "Pictures" "Videos" "Music")
+(defrules is ()
+  ((_ proc n)
+   (stx-number? #'n)
+   (lambda (v) (= n (proc v))))
+  ((_ proc c)
+   (stx-char? #'c)
+   (lambda (v) (eqv? c (proc v))))
+  ((_ proc s)
+   (stx-string? #'s)
+   (lambda (v) (string=? s (proc v))))
```

Comment:
this might fail with an exception if the result of proc is not a string; I think we should stick to `equal?` here.

Reply:
Or add  a test for string'iness, so `(let (v (proc v)) (and (string? v) (string=? s v)))`